src/main/java/uk/gov/justice/probation/courtcasematcher/service/MatcherService.java

```diff
@@ -53,8 +62,38 @@ public Mono<Defendant> matchDefendant(Defendant defendant, Hearing hearing) {
                     log.error(throwable.getMessage());
                     telemetryService.trackOffenderMatchFailureEvent(defendant, hearing);
                 })
+                .doOnSuccess(matchResponse -> enrichWithMatchScore(hearing, defendant, matchResponse))
                 .map(matchResponse -> HearingMapper.updateDefendantWithMatches(defendant, matchResponse))
                 ;
     }
 
+  private Mono<MatchResponse> enrichWithMatchScore(Hearing hearing, Defendant defendant, MatchResponse matchResponse) {
+      var matchRequest = matchRequestFactory.buildFrom(defendant);
+      var sourceDataset = PersonMatchScoreParameter.of(hearing.getSource().name(), "DELIUS");
+      if (matchResponse.getMatchCount() > 0) {
+
+        matchResponse.getMatches().stream().forEach(match -> {
+          PersonMatchScoreRequest request = buildPersonMatchScoreRequest(defendant, matchRequest, sourceDataset, match.getOffender());
+          personMatchScoreRestClient.match(request)
+            .doOnSuccess(personMatchScoreResponse -> match.setMatchProbability(personMatchScoreResponse.getMatchProbability().getValue0()))
+            .doOnError(throwable ->
+              log.warn("Error occurred while getting person match score for defendant id {}", defendant.getDefendantId(), throwable)
+            )
+            .block();
+        });
+      }
+      return Mono.just(matchResponse);
```

Comment:
This could return a `Mono.error()` which will cause the rest of the flow to fail I think. I think we need an `onErrorReturn(Mono.empty())` or something here.

Reply:
removed return statement as obsolete